drivers/vulkan/vulkan_context.cpp

```diff
@@ -1279,6 +1283,10 @@ Error VulkanContext::_create_physical_device(VkSurfaceKHR p_surface) {
 	//  features based on this query
 	vkGetPhysicalDeviceFeatures(gpu, &physical_device_features);
 
+	// Check required features
+	ERR_FAIL_COND_V_MSG(!physical_device_features.imageCubeArray, ERR_CANT_CREATE, "Your GPU doesn't support image cube arrays which are required to use the Vulkan-based renderers in Godot.");
+	ERR_FAIL_COND_V_MSG(!physical_device_features.independentBlend, ERR_CANT_CREATE, "Your GPU doesn't support independentBlend which is required to use the Vulkan-based renderers in Godot.");
+
```

Comment:
One more to solve https://github.com/godotengine/godot/issues/70013

We are getting the crash because the device doesn't support using arrays of samplers, the Molten VK error comes from here:
https://github.com/KhronosGroup/MoltenVK/blob/4fded82c679fcd8f743e9a86f46904ca9e85629c/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm#L647-L649

MoltenVK passes the presence of that feature using ``shaderSampledImageArrayNonUniformIndexing`` in the Vulkan Physical Device Features struct.

```suggestion
ERR_FAIL_COND_V_MSG(!physical_device_features.shaderSampledImageArrayNonUniformIndexing, ERR_CANT_CREATE, "Your GPU doesn't support shaderSampledImageArrayNonUniformIndexing which is required to use the Vulkan-based renderers in Godot.");

```

Reply:
Hmm, should we be checking both individually then, or should it be a matter of only failing if both are not available?

I really need to start getting back into Mac, I have a Mac now to test but so used to working on Windows now :P

Reply:
I don't think MoltenVK exposed any means for us to check them separately.

Reply:
Ok this fails, it's actually not in `VkPhysicalDeviceFeatures`, it's in `VkPhysicalDeviceVulkan12Features`. We'll need to add retrieving that data for our device (when possible) as well. This does mean that MoltenVK is guaranteed to support Vulkan 1.2?

Reply:
Looking further into `VkPhysicalDeviceVulkan12Features` it's a core function of Vulkan 1.2 and it actually replaces some of the others.

Reply:
Ok I changed it around so we use `VkPhysicalDeviceVulkan12Features` if available. There is probably more improvement possible here, I'm wondering for instance if we should cache the results better but thats for another day.

For now I've got the check for `shaderSampledImageArrayNonUniformIndexing` only applied when we do have Vulkan 1.2 support and only for MacOS. 
Not sure if we should do more for iOS and if we've not on Vulkan 1.2?

Reply:
It should be for macOS and iOS as https://github.com/godotengine/godot/issues/70013 was a report for iOS

But I think other than that our hands are tied as MoltenVK doesn't really give us anything better to my knowledge

Reply:
It should be tied into a vendor extension that we can check. Can't find anything on Google about it but if we find out if Metal announces itself as an extension we could check that.

Reply:
Checking the MoltenVK headers it identifies itself as a Vulkan 1.2.x implementation and they expose an extension `VK_MVK_MOLTENVK_EXTENSION_NAME` (`VK_MVK_moltenvk`) so we can probably check that.

I really need to take the time to setup my mac for testing

Reply:
Pushing to a further PR